library/src/test/java/com/novoda/downloadmanager/ServiceNotificationDispatcherTest.java

```diff
@@ -44,70 +34,122 @@
     @Before
     public void setUp() {
         Log.setShowLogs(false);
-        given(notificationCreator.createNotification(any(DownloadBatchStatus.class))).willReturn(SINGLE_PERSISTENT_NOTIFICATION_INFORMATION);
+        given(notificationCreator.createNotification(any(DownloadBatchStatus.class))).willReturn(createNotificationInfo(SINGLE_PERSISTENT_NOTIFICATION, 100));
 
         notificationDispatcher = new ServiceNotificationDispatcher<>(lock, notificationCreator, notificationManager);
         notificationDispatcher.setService(downloadService);
     }
 
     @Test
     public void showsSinglePersistentNotification() {
-        given(notificationCreator.createNotification(DOWNLOAD_BATCH_STATUS)).willReturn(SINGLE_PERSISTENT_NOTIFICATION_INFORMATION);
+        NotificationInformation notificationInfo = createNotificationInfo(SINGLE_PERSISTENT_NOTIFICATION, 100);
+        given(notificationCreator.createNotification(DOWNLOAD_BATCH_STATUS)).willReturn(notificationInfo);
 
         notificationDispatcher.updateNotification(DOWNLOAD_BATCH_STATUS);
 
-        verify(downloadService).start(SINGLE_PERSISTENT_NOTIFICATION_INFORMATION.getId(), SINGLE_PERSISTENT_NOTIFICATION_INFORMATION.getNotification());
+        verify(downloadService).start(notificationInfo.getId(), notificationInfo.getNotification());
     }
 
     @Test
     public void stacksDismissibleNotification() {
-        given(notificationCreator.createNotification(DOWNLOAD_BATCH_STATUS)).willReturn(STACKABLE_DISMISSIBLE_NOTIFICATION_INFORMATION);
+        NotificationInformation notificationInfo = createNotificationInfo(STACK_NOTIFICATION_DISMISSIBLE, 100);
+        given(notificationCreator.createNotification(DOWNLOAD_BATCH_STATUS)).willReturn(notificationInfo);
+
+        notificationDispatcher.updateNotification(DOWNLOAD_BATCH_STATUS);
+
+        InOrder inOrder = inOrder(downloadService, notificationManager);
+        inOrder.verify(downloadService, never()).stop(true);
```

Comment:
Do you need the `inOrder` here? You're checking for a `never` and then for a single call. Wouldn't it be the same if they were not in an `inOrder`?

Reply:
Yeah, in the tests with the `never` the `inOrder` can be omitted... will remove

Reply:
797d951